qiskit/optimization/algorithms/grover_minimum_finder.py

```diff
@@ -0,0 +1,309 @@
+                v = outcome[n_key:n_key + n_value]
+
+                # Convert the binary string to integer.
+                int_v = self._bin_to_int(v, n_value) + threshold
+                v = self._twos_complement(int_v, n_value)
+
+                self._logger.info('Iterations: {}'.format(rotation_count))
+                self._logger.info('Outcome: {}'.format(outcome))
+                self._logger.info('Value: {} = {}'.format(v, int_v))
+
+                # If the value is an improvement, we update the iteration parameters (e.g. oracle).
+                if int_v < optimum_value:
+                    optimum_key = k
+                    optimum_value = int_v
+                    self._logger.info('Current Optimum Key: {}'.format(optimum_key))
+                    self._logger.info('Current Optimum Value: {}'.format(optimum_value))
+                    if v.startswith("1"):
+                        improvement_found = True
+                        threshold = optimum_value
+                else:
+                    # No better number after the max number of iterations, so we assume the optimal.
+                    if loops_with_no_improvement >= self._n_iterations:
+                        improvement_found = True
+                        optimum_found = True
+
+                    # Using Durr and Hoyer method, increase m.
+                    # TODO: Give option for a rotation schedule, or for different lambda's.
+                    m = int(np.ceil(min(m * 8/7, 2**(n_key / 2))))
+                    self._logger.info('No Improvement. M: {}'.format(m))
+
+                # Check if we've already seen this value.
+                if k not in keys_measured:
+                    keys_measured.append(k)
+
+                # Stop if we've seen all the keys or hit the rotation max.
+                if len(keys_measured) == num_solutions or rotations >= max_rotations:
+                    improvement_found = True
+                    optimum_found = True
+
+                # Track the operation count.
+                operations = circuit.count_ops()
+                operation_count[iteration] = operations
+                iteration += 1
+                self._logger.info('Operation Count: {}\n'.format(operations))
+
+        # Get original key and value pairs.
+        func_dict[-1] = orig_constant
+        solutions = get_qubo_solutions(func_dict, n_key)
+
+        # If the constant is 0 and we didn't find a negative, the answer is likely 0.
+        if optimum_value >= 0 and orig_constant == 0:
+            optimum_key = 0
+        opt_x = [1 if s == '1' else 0 for s in ('{0:%sb}' % n_key).format(optimum_key)]
+
+        # Build the results object.
+        grover_results = GroverOptimizationResults(operation_count, rotations, n_key, n_value,
+                                                   func_dict)
+        result = OptimizationResult(x=opt_x, fval=solutions[optimum_key],
+                                    results=grover_results)
```

Comment:
@AustinGilliam If the GroverOptimizationResults don't derive from the OptimizationResult that could be confusing. Also, if the user gets back `result` he has to re-access the grover results and that throws the typing a bit overboard.

Could we do it as follows?
```
result = GroverOptimizationResult(x=x_opt, fval=solutions[optimum_key], operation_count=operation_count,
rotations=rotations, n_key=n_key, n_value=n_value, func_dict=func_dict)
```
and specify `GroverOptimizationResult` such that it fits your algorithm
```
class GroverOptimizationResult(OptimizationResult):
   # add required properties operation_count, rotations, etc.
```

Reply:
I tried this out, but it conflicts with several other parts of the library, such as the integer_to_binary_converter.py, which assumes the format used currently (x, fval, and then everything else dumped into results).

Reply:
I adjusted the int_to_bin converter such that the "decode" method just changes the x value and does not create a new results object, with this adjustment it should work - or at least should be easy to fix it doesn't work yet.

Reply:
Alright - I'll update and see if that works.